### ISSUE TITLE
Testmodus mit Fahrzeugen

### ISSUE THREAD

**Issue by atemnot** [Stephan-S/FS22_AutoDrive#272]
Hallo zusammen,

für das Testen von Streckennetzen wäre es sehr Hilfreich, wenn es einen Testmodus gäbe um die Strecken im Livebetrieb zu testen.
Ich würde mir vorstellen, dass man Zielpunkte, die angefahren werden sollen, in eine Liste einfügt (vielleicht im Debugmenü). Danach versetzt man Fahrzeuge und Gespanne in einen Testmodus. In diesem fahre sie in Endlosschleife entweder zufällige Punkte aus der Liste oder die Punkte in der Reihenfolge der Liste (Auswahl des Testmodus) an. 
Zusätzlich sollte es bei Problemen (z.B. dass ein Fahrer stecken bleibt bzw. zu lange warten muss) eine Warnmeldung geben, damit man sich die Stelle ansehen kann.
Im Moment wähle ich bei einigen Testfahrzeugen manuell zufällige Ziele und lasse sie mit verschiedenen Traktor- und Wagengrößen fahren. Das ist aber aufwändig und man kann in der Zeit nichts andere machen.

**Comment by Iwan1803** [comment]
Wenn Fahrzeuge an Wänden lang schrammen, würdest du das so ja gar nicht mitbekommen.
Ebenso würdest du auch nicht merken, wenn sie einen Umweg nehmen, weil z.Bsp irgendwo eine Abbiegung fehlt.
Ein Testmodus würde dir also lediglich aufzeigen, wenn sie wirklich irgendwo hängen bleiben.
Für fehlerhafte Wegstrecken hat AD ja schon eine Debug-Modus.
Was also würde ein "Testmodus" für einen gravierenden Vorteil bringen, außer das du deine Testfahrzeuge nicht händisch losschicken musst?

**Comment by atemnot** [comment]
Es geht mir um die Interaktion zwischen zwei Fahrzeugen z.B. für den Betrieb mit mehreren Abfahrern.
Die Kollision mit statischen Objekten kann ich ja mit einer Testfahrt überprüfen. Einfach mit dem größtmöglichen Gespann abfahren und einmal ansehen.
Was ich nicht prüfen kann ist, ob die Abstände zwischen zwei Strecken immer ausreichen (generell aber auch insbesondere bei Kurven. Oder ob Ausweichstrecken lang genug sind, wie sich bestimmte Streckenabschnitte (z.B. nahe Abladestationen) bei hohem Verkehrsaufkommen verhalten.
Ich würde dafür gerne zwei oder mehr Gespanne mit leicht unterschiedlicher Geschwindigkeit automatisch zwischen zwei Zielpunkten pendeln lassen, damit sie sich an immer anderen Stelle begegnen. Klar wäre damit auch keine 100%ige Absicherung gegeben, aber ich würde viele Begegnungsmöglichkeiten automatisch testen können. Das ist manuell eigentlich gar nicht möglich.
Es geht mir nicht darum zu prüfen ob die Strecken ansich funktionieren oder fehlende Verbindungen oder so haben. Das ist ja schon möglich über den Debug-Modus oder auch mit der Streckenanzeige auf der Job-Karte.

**Comment by marc950** [comment]
ja, stimmt das Problem habe ich auch: wenn ich mehre Gespanne fahren lassen bleiben sie in der Kurve stehen ode Kreuzungen gegenläufige Fahrer.
